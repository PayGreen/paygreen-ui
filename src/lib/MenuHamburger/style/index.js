import styled from 'styled-components';
import { math } from 'polished';
import { line, sideStyle, isClosedStyle, isOpenStyle } from './base';
import { mainColor } from '../../Text/style/constants';

const MenuHamburgerBase = styled.button`
    display: block;
    box-sizing: border-box;
    width: ${props => math(props.theme.icon.size.lg + '-' + line)};
    height: ${props => math(props.theme.icon.size.lg + '-' + line)};
    background-color: transparent;
    border: none;
    padding: 0;
    outline: none;
    cursor: pointer;

    span {
        display: block;
        height: ${line}px;
        border-radius: ${props => props.theme.radius.sm};
        background-color: ${props => mainColor[props.colorPallet]};
        transition: all ${props => props.theme.transition.sm};
        ${props => sideStyle[props.arrowDirection]};

        &:nth-of-type(2) {
            margin-top: ${line * 2}px;
            margin-bottom: ${line * 2}px;
        }
    }

    ${props => (props.isOpen ? isOpenStyle : isClosedStyle)};
`;

export { MenuHamburgerBase };
