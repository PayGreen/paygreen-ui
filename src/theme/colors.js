const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
