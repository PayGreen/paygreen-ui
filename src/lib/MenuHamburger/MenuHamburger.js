import React from 'react';
import PropTypes from 'prop-types';
import {
    arrowDirectionOptions,
    arrowDirectionDefault,
    colorPalletOptions,
    colorPalletDefault,
    colorThemeOptions,
    colorThemeDefault,
    formStatusOptions,
    formStatusDefault,
    greyOptions,
    greyDefault,
    spaceOptions,
    spaceDefault
} from '../../shared/constants';
import { MenuHamburgerBase } from './style';

const MenuHamburger = props => (
    <MenuHamburgerBase {...props}>
        <span></span>
        <span></span>
        <span></span>
    </MenuHamburgerBase>
);

MenuHamburger.propTypes = {
    arrowDirection: PropTypes.oneOf(Object.values(arrowDirectionOptions)),
    isOpen: PropTypes.bool,
    colorPallet: PropTypes.oneOf(Object.values(colorPalletOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    colorWab: PropTypes.oneOf(Object.values(greyOptions)),
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginRight: PropTypes.oneOf(Object.values(spaceOptions)),
    marginLeft: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
};

MenuHamburger.defaultProps = {
    arrowDirection: arrowDirectionDefault,
    isOpen: false,
    colorPallet: colorPalletDefault,
    colorTheme: colorThemeDefault,
    colorWab: greyDefault,
    colorStatus: formStatusDefault,
    marginTop: spaceDefault,
    marginRight: spaceDefault,
    marginLeft: spaceDefault,
    marginBottom: spaceDefault,
};

export default MenuHamburger;
