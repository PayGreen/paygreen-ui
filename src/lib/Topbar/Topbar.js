import React from 'react';
import { TopbarBase } from './style';

const Topbar = props => {
    return <TopbarBase {...props}>{props.children}</TopbarBase>;
};

export default Topbar;