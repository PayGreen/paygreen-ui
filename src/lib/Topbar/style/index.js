import styled from 'styled-components';
import { topbarStyle } from './base';
import { LogoBase } from '../../Logo/style';

const TopbarBase = styled.header`
    .main-nav {
        a {
            display: block;
            height: 100%;
        }
    }

    @media (${props => props.theme.query.max.md}) {
        .main-nav {
            ${topbarStyle};
        }

        ${LogoBase} {
            padding: 0 !important;
        }
    }

    @media (${props => props.theme.query.min.md}) {
        .main-nav {
            ${topbarStyle};
        }

        ${LogoBase} {
            display: none;
        }
    }
`;

export { TopbarBase };
