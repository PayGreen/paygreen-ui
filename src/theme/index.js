import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
