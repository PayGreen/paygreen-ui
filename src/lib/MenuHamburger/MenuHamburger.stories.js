import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, radios, boolean, select } from '@storybook/addon-knobs';
import {
    folder,
    colorThemeOptions,
    colorThemeDefault,
    colorPalletOptions,
    colorPalletDefault,
    greyOptions,
    greyDefault,
} from '../../shared/constants';
import MenuHamburger from './MenuHamburger';

const { status, ...hamburgerColorPalletOptions } = colorPalletOptions;

storiesOf(folder.nav + folder.sub.control + 'MenuHamburger', module)
    .addDecorator(withKnobs)
    .add('MenuHamburger', () => (
        <MenuHamburger
            colorPallet={radios(
                'Color pallet',
                hamburgerColorPalletOptions,
                colorPalletDefault,
            )}
            colorTheme={select(
                'Color theme',
                colorThemeOptions,
                colorThemeDefault,
            )}
            colorWab={select('Color wab', greyOptions, greyDefault)}
            isOpen={boolean('Is open', false)}
        />
    ));
