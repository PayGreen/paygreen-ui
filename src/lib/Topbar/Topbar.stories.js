import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import { folder } from '../../shared/constants';
import MenuHamburger from '../MenuHamburger/MenuHamburger';
import Topbar from './Topbar';

storiesOf(folder.nav + 'Topbar', module)
    .addDecorator(withKnobs)
    .add('Topbar', () => (
        <Topbar>
            <MenuHamburger isOpen={boolean('Is open', true)} />
        </Topbar>
    ));
