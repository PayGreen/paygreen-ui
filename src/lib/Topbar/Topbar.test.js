import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import MenuHamburger from '../MenuHamburger/MenuHamburger';
import Topbar from './Topbar';

it('renders without crashing', () => {
    const topbar = TestRenderer.create(
        <Topbar theme={ThemeDefault}>
            <div className="main-nav">
                <MenuHamburger theme={ThemeDefault} />
            </div>
        </Topbar>,
    );
    expect(topbar.toJSON()).toMatchSnapshot();
});
