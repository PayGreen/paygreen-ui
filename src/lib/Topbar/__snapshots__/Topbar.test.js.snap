// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  className="style__TopbarBase-sc-167w4io-0 bgLEjQ"
>
  <button
    className="style__MenuHamburgerBase-sc-1tooz6t-0 czVrHi"
    type="button"
  >
    <span />
    <span />
    <span />
  </button>
</header>
`;
